Build failure: globalprotect-openconnect

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. Use latest rev of Unstable (25.11)

2. Run
    ```bash
    nix-shell -p globalprotect-openconnect
    ```

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
$nix log /nix/store/r405m8xh0c5bwv3qdqqszf0s1rlpzssh-globalprotect-openconnect-1.4.9.drv
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/kgh37c6xg8k3z5rb1b484fi6z25dwym9-globalprotect-openconnect-1.4.9.tar.gz
source root is globalprotect-openconnect-1.4.9
setting SOURCE_DATE_EPOCH to timestamp 1673182977 of file "globalprotect-openconnect-1.4.9/3rdparty/qtkeychain/translations/translations.qrc.in"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation globalprotect-openconnect-1.4.9: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'GPService/gpservice.h')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/w1ighp07fyjqwsc8v11vj9khhn4knj8d-globalprotect-openconnect-1.4.9/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/w1ighp07fyjqwsc8v11vj9khhn4knj8d-globalprotect-openconnect-1.4.9/libexec -DCMAKE_INSTALL_LI>
-- The CXX compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_INSTALL_PREFIX was set to: /nix/store/w1ighp07fyjqwsc8v11vj9khhn4knj8d-globalprotect-openconnect-1.4.9
CMake Error at 3rdparty/qt-unix-signals/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

-- Configuring incomplete, errors occurred!
```

### Additional context

https://github.com/NixOS/nixpkgs/issues/365050
https://github.com/NixOS/nixpkgs/pull/355758

### System metadata

**Platform:**  x86_64-linux

### Notify maintainers

@jerith666 
@Binary-Eater 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/networking/globalprotect-openconnect/default.nix
+++ b/pkgs/tools/networking/globalprotect-openconnect/default.nix
@@ -24,6 +24,10 @@ stdenv.mkDerivation rec {
     wrapQtAppsHook
   ];
 
+  cmakeFlags = [
+    (lib.cmakeFeature "CMAKE_POLICY_VERSION_MINIMUM" "3.16")
+  ];
+
   buildInputs = [
     openconnect
     qtwebsockets
@@ -33,9 +37,11 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     substituteInPlace GPService/gpservice.h \
-      --replace /usr/local/bin/openconnect ${openconnect}/bin/openconnect;
+      --replace-fail /usr/local/bin/openconnect ${openconnect}/bin/openconnect;
     substituteInPlace GPService/CMakeLists.txt \
-      --replace /etc/gpservice $out/etc/gpservice;
+      --replace-fail /etc/gpservice $out/etc/gpservice;
+    # Force minimum CMake version to 3.16 to avoid policy warnings
+    find . -name "CMakeLists.txt" -exec sed -i 's/cmake_minimum_required(VERSION [^)]*)/cmake_minimum_required(VERSION 3.16)/g' {} \;
   '';
 
   meta = with lib; {

